app/response_svc.py

```diff
@@ -85,25 +85,30 @@ async def refresh_blue_agents_abilities(self):
             if a not in self.abilities:
                 self.abilities.append(a)
 
-    async def run_abilities_on_agent(self, blue_agent, original_pid):
+    async def run_abilities_on_agent(self, blue_agent, red_agent, original_pid):
         facts = [Fact(trait='host.process.id', value=original_pid)]
         links = []
+        relationships = []
         for ability_id in self.abilities:
-            ability_facts, ability_links = await self.run_ability_on_agent(blue_agent, ability_id, facts, original_pid)
+            ability_facts, ability_links, ability_relationships = await self.run_ability_on_agent(blue_agent, red_agent, ability_id, facts, original_pid, relationships)
```

Comment:
if you are just using str(red_agent.pid) then just use that here red_agent_pid instead of the whole agent. sending two agents to run_ability_on_agent makes it sort of ambiguous as to which agent the function is taking action on